### DISCUSSION TITLE
Need help with src_fmt_configs for parquet files

### DISCUSSION THREAD

**Discussion by dilex42** [apache/airflow#19692] [Q&A]
Hi. I have a small issue here and wanted to ask for advice. I want to load parquet file to BigQuery but I have a column that is a list of strings. When I do default insertion using GCSToBigQueryOperator or BigQueryInsertJobOperator, BQ autodefines my special column as record of repeated records instead of repeated string field. I found solution in the form of `"parquetOptions": {"enableListInference": True}` option added to configuration in BigQueryInsertJobOperator:
```
BigQueryInsertJobOperator(
        task_id="load_data",
        configuration={
            "load": {
                "writeDisposition": "WRITE_TRUNCATE",
                "createDisposition": "CREATE_IF_NEEDED",
                "destinationTable": {
                    "projectId": "my_project",
                    "datasetId": "my_dataset",
                    "tableId": "my_table",
                },
                "sourceUris": [f"gs://{my_bucket}/{file_path}"],
                "sourceFormat": "PARQUET",
                "parquetOptions": {"enableListInference": True},
            }
        },
    )
```
Now, my problem is that I would like to use GCSToBigQueryOperator if it's possible. And while I can map most configuration options, I can't add `enableListInference` to parquet options.
```
GCSToBigQueryOperator(
        task_id="load_data",
        bucket=my_bucket,
        destination_project_dataset_table=f"{my_project}.{my_dataset}.{my_table}",
        source_objects=[file_path],
        source_format="PARQUET",
        write_disposition="WRITE_TRUNCATE",
        create_disposition="CREATE_IF_NEEDED",
    )
```
After studying source code I found that there is an option to add config options to GCSToBigQueryOperator like this
```
src_fmt_configs={"enableListInference": True}
```
but according to [this line](https://github.com/apache/airflow/blob/2976070ebe441b1ebed60cceceaeac20a6f4a37a/airflow/providers/google/cloud/hooks/bigquery.py#L1891) `enableListInference` is not a valid config for parquet files. And I can't understand why it's not an option there. For example similar Avro option `useAvroLogicalTypes` is there.

So my question – would simply adding `enableListInference` to valid parquet options list would solve my problem(I am more than happy to add PR) or this option is not there because it will break something else(which I can't figure out). TY.

**Comment by dilex42** [comment]
up

**Comment by bhaskerallene** [comment]
I wonder If there is a workaround for this issue?
@dilex42 how did you get past this?

**Reply by dilex42** [reply]
Hi @bhaskerallene. We just use `BigQueryInsertJobOperator` as I wrote in first example.
